File: app/src/main/java/fr/free/nrw/commons/media/MwParseResponse.java
package fr.free.nrw.commons.media;

import androidx.annotation.Nullable;
import androidx.annotation.VisibleForTesting;

import org.wikipedia.dataclient.mwapi.MwResponse;

public class MwParseResponse extends MwResponse {
    @Nullable
    private MwParseResult parse;

    @Nullable
    public MwParseResult parse() {
        return parse;
    }

    public boolean success() {
        return parse != null;
    }

    @VisibleForTesting
    protected void setParse(@Nullable MwParseResult parse) {
        this.parse = parse;
    }
}

File: app/src/main/java/fr/free/nrw/commons/media/MwParseResponse.kt
package fr.free.nrw.commons.media

import androidx.annotation.VisibleForTesting
import fr.free.nrw.commons.wikidata.mwapi.MwResponse

class MwParseResponse : MwResponse() {
    private var parse: MwParseResult? = null

    fun parse(): MwParseResult? = parse

    fun success(): Boolean = parse != null

    @VisibleForTesting
    protected fun setParse(parse: MwParseResult?) {
        this.parse = parse
    }
}
